docs/build/smart-contracts/language/README.md

```diff
@@ -151,6 +153,72 @@ Since the maps and the lists have a very close syntax, it is impossible to diffe
 The brackets are mostly optional! `key1: 1, key2: 2` will work as well. But if you have a nested map, you cannot omit them!
 :::
 
+## Functions
+
+In the Archethic Smart Contract Language, you can declare 2 types of functions:
+
+- Internal Functions declared with the `fun` keyword
+- External Functions declared with the `export fun` keyword
+
+### Internal Functions
+
+Internal functions are functions that are only available in the Smart Contract. They are not callable from the outside.
+
+They are declared with the `fun` keyword and can be called from the [Action](/build/smart-contracts/language/actions) or [Condition](/build/smart-contracts/language/condition) blocks.
+
+They can have 0 argument :
+
+```elixir
+fun hello() do
+    "Hello World!"
+end
+```
+
+Or more :
+
+```elixir
+fun sum(a, b) do
+    a + b
+end
+```
+
+**These functions aren't able to call another internal function** but can call an external one.
+
+:::info
+Internal functions are able to use [library module](#library) IO functions. IO stands for Input/Output. These functions are used to fetch data from the blockchain.
+:::
+:::warning
+Internal functions are not able to mutate the state of the Smart Contract.
```

Comment:
Same here with caution.
Also I think "mutate the state" is not a good term since we could understand it as the real state of a smart contract and not the output transaction.
So maybe you could say both like "Internal functions cannot modify the Smart Contract's state or update the next Contract transaction"